core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1241,6 +1243,83 @@ public static Expression multiplyDivide(Expression e, BigDecimal multiplier,
     }
   }
 
+  /** Implementor for the {@code MODE} aggregate function. */
+  static class ModeImplementor extends StrictAggImplementor {
+    @Override protected void implementNotNullReset(AggContext info,
+        AggResetContext reset) {
+      // acc[0] = null;
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(0),
+                  Expressions.constant(null))));
+      // acc[1] = new HashMap<>();
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(1),
+                  Expressions.new_(HashMap.class))));
+      // acc[2] = Long.valueOf(0);
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(2),
+                  Expressions.constant(0, Long.class))));
+    }
+
+    @Override protected void implementNotNullAdd(AggContext info,
+        AggAddContext add) {
+      Expression currentArg = add.arguments().get(0);
```

Comment:
Nit: this will be very useful if you can also add equivalent code/pesudo code into comment.

Reply:
arguments method just can get the column value. So I don't add a comment